### PR TITLE
GitHub Flow Migration

### PR DESCRIPTION
**Pull request by ddsjoberg**
**The devel branch here still has the release version matching the CRAN release. Needs to be updated before merging!**

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/git_usage.Rmd [hunk 1]
```diff
@@ -43,23 +38,23 @@ Each feature branch must be related to an issue. We encourage new developers to
 
 ### Naming Branches
 
-The name of the branch must be prefixed with the issue number, followed by a short but meaningful description and the `@<target_branch>` suffix. The latter would be `@devel` in most cases. As an example, given an issue #94 "Program function to derive `LSTALVDT`", the branch name would be `94_derive_var_lstalvdt@devel`.
+The name of the branch must be prefixed with the issue number, followed by a short but meaningful description and the `@<target_branch>` suffix. The latter would be `@main` in most cases. As an example, given an issue #94 "Program function to derive `LSTALVDT`", the branch name would be `94_derive_var_lstalvdt@main`.
```
Comment (bundfussr) [inline vignettes/git_usage.Rmd:41]: `@main` does not need to be added as it is the fall back branch.

File: vignettes/release_strategy.Rmd [hunk 2]
```diff
@@ -41,37 +41,29 @@ updates via the community meetings, especially any anticipated breaking changes.
 
 ## Quarterly Release
 
-A package release is done in five parts:
+A package release is done in four parts:
 
-1) Create a Pull Request from `devel` into the `main` branch.  Issues identified in this Pull Request should have work done in separate branches and merged once again into `devel`.
-1) Verify that all CI/CD checks are passing for the `devel` into the `main` Pull Request, merge and then bundle up and send off to CRAN. See the [chapter](https://r-pkgs.org/release.html#decide-the-release-type) in R Packages for more details.
+1) Verify that all CI/CD checks and then bundle up and send off to CRAN. See the [chapter](https://r-pkgs.org/release.html#decide-the-release-type) in R Packages for more details.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:46]: Don't we need a PR into `main` were we update the version number (remove `.9000` part and increase version)?

File: vignettes/release_strategy.Rmd [hunk 3]
```diff
@@ -41,37 +41,29 @@ updates via the community meetings, especially any anticipated breaking changes.
...
 1) If CRAN asks for modifications, repeat steps 1-2 as necessary.
 1) Once the package is accepted and available on CRAN, a [GitHub action](https://github.com/pharmaverse/admiral/actions/workflows/pages/pages-build-deployment) is set up to rebuild the `{admiral}` website with all the updates for this release.
 1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
-1) Any issues fixed in the `main` branches should be merged back into `devel`.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:51]: Do we need a PR to `main` which adds the `.9000` again or should this be done in the first feature branch?

File: vignettes/release_strategy.Rmd [hunk 4]
```diff
@@ -41,37 +41,29 @@ updates via the community meetings, especially any anticipated breaking changes.
...
 ## Hot Fix Release
 
-Occasionally we will need to release a hot fix to address a package breaking bug.  A hot fix release is done in 6 parts:
+Occasionally we will need to release a hot fix to address a package breaking bug.  A hot fix release is done in eight parts:
 
 1) Identify all the bugs that need to be fixed for this hot fix release and label with hot fix label.
-1) Branches addressing the bugs should have Pull Requests merged into the `patch` branch **NOT** the `devel` branch.
-1) When naming the branch follow the [naming conventions](git_usage.html#implementing-an-issue) guide but use the `@main` suffix 
+1) Branches addressing the bugs should have Pull Requests merged into a single `patch` branch **NOT** the `main` branch, where the `patch` branch has been created from the most recent release of the package.
+1) When naming the branch follow the [naming conventions](git_usage.html#implementing-an-issue) guide but use the `@main` suffix.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:61]: I think the `@main` suffix must not be used because it refers to the devel version of the upstream dependencies. Instead we must use the latest released version of the upstream dependencies. I am not sure how we can refer to them.

**Comment by ddsjoberg** [comment]
@bundfussr feel free to make any changes you see fit. I am less familiar with staged dependencies, and wasn't sure which notation is required there.

File: vignettes/release_strategy.Rmd (hunk 3 above)
Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:51]: When we make releases, we should follow the checklist issue created from `usethis::use_release_issue()`. The last thing to do there is bump the version to `xxx.9000`. There is no need to wait until the first PR to increase it (although this is certainly something i have done in the past, and either way works perfectly well. But good to stick with standard practices when we can)

Comment (bundfussr) [reply vignettes/release_strategy.Rmd:51]: Should we use `use_release_issue()` and follow the list created by the call or should we follow our list? Unfortunately I couldn't find an example of the list created by `use_release_issue()` in the usethis documentation. Most likely there are many overlaps. I think we should have just one list to follow.

File: vignettes/release_strategy.Rmd [hunk 5]
```diff
@@ -41,37 +41,29 @@ updates via the community meetings, especially any anticipated breaking changes.
...
-1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
-1) These hot fixes should then be merged into the `devel` branch through an additional Pull Request.  
-
-**Hot Fix Release**: `patch >>  main  >> devel`
+1) Create a GitHub release on the `patch` branch. Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
+1) These hot fixes should then be merged into the `main` branch through an additional Pull Request.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:66]: We need to add that the development version needs to be added again in `main`.

File: vignettes/release_strategy.Rmd [hunk 6]
```diff
@@ -41,37 +41,29 @@ updates via the community meetings, especially any anticipated breaking changes.
...
 
-Be on the look out for an automated check PR whenever `devel` goes into `main` as a future upcoming feature.
-
 ## Hot Fix Release
 
-Occasionally we will need to release a hot fix to address a package breaking bug.  A hot fix release is done in 6 parts:
+Occasionally we will need to release a hot fix to address a package breaking bug.  A hot fix release is done in eight parts:
 
 1) Identify all the bugs that need to be fixed for this hot fix release and label with hot fix label.
-1) Branches addressing the bugs should have Pull Requests merged into the `patch` branch **NOT** the `devel` branch.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:66]: We should add that the `patch` branch is updated by merging the latest released version into `patch`.

File: vignettes/git_usage.Rmd [hunk 7]
```diff
@@ -24,14 +24,9 @@ This article will give you an overview of how the `{admiral}` project is utilizi
 
 # Branches
 
--	The `main` branch contains the latest **released** version and should not be used for development. You can find the released versions [here](https://GitHub.com/pharmaverse/admiral/releases) 
--	The `devel` branch contains the latest development version of the package. You will always be branching off and pulling into the `devel` branch. This is set as the default branch on GitHub.
+-	The `main` branch contains the latest development version of the package.
 - The `gh-pages` branches contains the code used to render this website you are looking at right now!
-- The `patch` branch is reserved for special hot fixes to address bugs. More info in [Hot Fix Release](release_strategy.html#hot-fix-release)
-- The `main`, `devel`, `gh-pages`, `patch` branches are under protection. If you try and push changes to these branches you will get an error unless you are an administrator.
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply vignettes/git_usage.Rmd:31]: added back (but in a new PR). All comments are regarding the new PR

File: vignettes/git_usage.Rmd (hunk 1 above)
Comment (ddsjoberg) [reply vignettes/git_usage.Rmd:41]: removed the detail about main as it's not needed

File: vignettes/release_strategy.Rmd (hunk 2 above)
Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:46]: that depends on the workflow one uses. the PR checklist from usethis, has you keep your version bump to the release version local until CRAN accepts. But others do different things.

File: vignettes/release_strategy.Rmd (hunk 4 above)
Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:61]: updated
